src/components/TreeView.js

```diff
@@ -56,9 +59,44 @@ function RowValue(props) {
     separator = ': ';
   }
 
+  if (position === 'TransactionEnvelope.signatures') {
+    checkSignatures({signatures: node, fetchedSigners});
+    return formatSignatureCheckState({node, separator})
+  }
+
+  if (position.match(/^TransactionEnvelope\.signatures\[[0-9]+\]\.signature$/)) {
+    return formatSignature({node, separator});
+  }
+
   return <span><strong>{node.type}</strong>{separator}{value}</span>
 }
 
+function checkSignatures({signatures, fetchedSigners}) {
+  // catch fetch signature errors
+  if (fetchedSigners === null || fetchedSigners === "PENDING") {
+    signatures.state = SIGNATURE.NOT_VERIFIED_YET;
+    return
+  } else if (fetchedSigners === "ERROR") {
+    signatures.state = SIGNATURE.INVALID;
+    return
+  }
+  
+  signatures.state = SIGNATURE.VALID;
+
+  for (var i = 0; i < signatures.nodes.length; i ++) {
+    const sig = signatures.nodes[i].nodes[1];
```

Comment:
This can be confusing if you don't remember that `signatures.nodes` are [decorated signatures](https://github.com/stellar/stellar-core/blob/master/src/xdr/Stellar-transaction.x#L10). Let's change it to:
```js
const sig = signatures.nodes[i].find(n => n.type == 'signature');
```